plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -129,10 +131,12 @@ func (c *X509Cert) getCert(u *url.URL, timeout time.Duration) ([]*x509.Certifica
 
 		hsErr := conn.Handshake()
 		if hsErr != nil {
+			c.tlsCfg.ServerName = "" // reset SNI, otherwise we reuse c.tlsCfg.ServerName
```

Comment:
I think you better make this a
```golang
defer func() { c.tlsCfg.ServerName = "" }()
```
maybe before doing the handshake in order to be prepared for future "exit-on-error" additions.

Reply:
Thanks, this seems less error prone. I changed this with commit 858b677